src/tekton/webviewstartpipeline.ts

```diff
@@ -0,0 +1,101 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { Command } from '../tkn';
+import { TektonItem } from './tektonitem';
+
+export interface Param {
+  name: string;
+  type?: string;
+}
+
+export interface PipelineParam extends Param {
+  default?: string | string[];
+  description?: string;
+}
+
+export interface PipelineResource {
+  name: string;
+  type: string;
+}
+
+export interface PipelineWorkspace extends Param {
+  type?: string;
+  workspace?: string;
+  data?: {
+    [key: string]: string;
+  };
+}
+
+
+export interface PipelineTaskRef {
+  kind?: string;
+  name: string;
+}
+
+export interface PipelineTaskParam {
+  name: string;
+  value: any;
+}
+
+export interface PipelineTaskResource {
+  name: string;
+  resource?: string;
+  from?: string[];
+}
+
+export interface PipelineTaskResources {
+  inputs?: PipelineTaskResource[];
+  outputs?: PipelineTaskResource[];
+}
+
+export interface PipelineTask {
+  name: string;
+  runAfter?: string[];
+  taskRef: PipelineTaskRef;
+  params?: PipelineTaskParam[];
+  resources?: PipelineTaskResources;
+  workspaces?: PipelineWorkspace[];
+}
+
+interface TknResource {
+  pipeline: {
+    spec: {
+      params?: PipelineParam[];
+      resources?: PipelineResource[];
+      workspaces?: PipelineWorkspace[];
+      tasks: PipelineTask[];
+      serviceAccountName?: string;
+    };
+  };
+  pipelineResource: any;
+  Secret: any;
+  ConfigMap: any;
+  PersistentVolumeClaim: any;
+}
+
+
+export const pipelineData = async (name: string) => {
```

Comment:
I think you can define function instead of exporting constant, which you assign lambda, or you have particular reason to do that?

Reply:
No particular reason for using lambda. I will use function here.